PWGHF/TableProducer/candidateSelectorDstar.cxx

```diff
@@ -0,0 +1,285 @@
+  /// @return true/false depending on success of selection
+  template <typename T1, typename T2>
+  bool selectionTopolConjugate(const T1& d0candidate, const T2& dstarCandidate)
+  {
+    auto dstarpT = dstarCandidate.candDStarpt();
+    auto pTBin = findBin(dstarconfigurable.binsPtForDstar, dstarpT);
+    if (pTBin == -1) {
+      return false;
+    }
+    auto SoftPi = dstarCandidate.prongPi();
+    // auto prong0 = d0candidate.prong0();
+    // auto prong1 = d0candidate.prong1();
+
+    if (SoftPi.sign() > 0.) { // Selection of D*+
+      auto Mpipik = std::array{massPi, massPi, massK};
+      auto InvDstar = dstarCandidate.dstarInvMass(Mpipik);
+      auto Mpik = std::array{massPi, massK};
+      auto InvD0 = d0candidate.d0m(Mpik);
+      // LOGF(info,"+ %d,%f,%f",SoftPi.sign(),InvDstar,InvD0);
+      if (std::abs(InvDstar - InvD0) > dstarconfigurable.cutsForDstar->get(pTBin, "DeltaMDstar")) {
+        return false;
+      }
+      // cut on D0 daughter DCA - need to add secondary vertex constraint here
+      if (std::abs(d0candidate.impactParameter0()) > d0configurable.cutsForD0->get(pTBin, "d0pi") || std::abs(d0candidate.impactParameter1()) > d0configurable.cutsForD0->get(pTBin, "d0K")) {
+        return false;
+      }
+      // cut on cos(theta*)
+      if (std::abs(D0_from_Dstar::cosThetaStarD0(d0candidate)) > d0configurable.cutsForD0->get(pTBin, "cos theta*")) {
+        return false;
+      }
+
+    } else if (SoftPi.sign() < 0) { // Selection of D*-
+      auto Mpikpi = std::array{massPi, massK, massPi};
+      auto InvDstar = dstarCandidate.dstarInvMass(Mpikpi);
+      auto Mkpi = std::array{massK, massPi};
+      auto InvD0 = d0candidate.d0m(Mkpi);
+      // LOGF(info,"- %d,%f,%f",SoftPi.sign(),InvDstar,InvD0);
+      if (std::abs(InvDstar - InvD0) > dstarconfigurable.cutsForDstar->get(pTBin, "DeltaMDstar")) {
+        return false;
+      }
+      // cut on D0 daughter DCA - need to add secondary vertex constraint here
+      if (std::abs(d0candidate.impactParameter0()) > d0configurable.cutsForD0->get(pTBin, "d0K") || std::abs(d0candidate.impactParameter1()) > d0configurable.cutsForD0->get(pTBin, "d0pi")) {
+        return false;
+      }
+      // cut on cos(theta*)
+      if (std::abs(D0_from_Dstar::cosThetaStarD0bar(d0candidate)) > d0configurable.cutsForD0->get(pTBin, "cos theta*")) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  void process(aod::Tracks const&, aod::Hf2Prongs const&, aod::HfD0fromDstar const& rowsD0cand, aod::HfCandDStarBase const& rowsDstarCand)
+  {
+    LOG(info) << "selector: processQA called";
+    for (auto& DstarCand : rowsDstarCand) {
+      auto GI = DstarCand.globalIndex();
+      auto D0Cand = rowsD0cand.iteratorAt(GI);
+      auto result1 = selectionOnDstar(D0Cand, DstarCand);
+      auto result2 = selectionTopolConjugate(D0Cand, DstarCand);
+      LOGF(info, "Results of dSelectionOnDstar function = %d, %d", result1, result2);
+    }
+  }
+};
```

Comment:
In this function you still miss the production of the table with the response for each candidate